plugins/plugin-machine/che-plugin-machine-ext-server/src/main/java/org/eclipse/che/ide/ext/machine/server/ssh/KeysInjector.java

```diff
@@ -69,12 +70,26 @@ public void onEvent(MachineStatusEvent event) {
                     try {
                         final Instance machine = environmentEngine.getMachine(event.getWorkspaceId(),
                                                                               event.getMachineId());
+
+                        // get machine keypairs
                         List<SshPairImpl> sshPairs = sshManager.getPairs(machine.getOwner(), "machine");
-                        final List<String> publicKeys = sshPairs.stream()
+                        final List<String> publicMachineKeys = sshPairs.stream()
                                                              .filter(sshPair -> sshPair.getPublicKey() != null)
                                                              .map(SshPairImpl::getPublicKey)
                                                              .collect(Collectors.toList());
 
+                        // get workspace keypair
+                        SshPairImpl sshWorkspacePair = sshManager.getPair(machine.getOwner(), "workspace", event.getWorkspaceId());
+
+                        // build list of all pairs.
+                        final List<String> publicKeys;
+                        if (sshWorkspacePair != null && sshWorkspacePair.getPublicKey() != null) {
+                            publicKeys = new ArrayList<>(publicMachineKeys.size() + 1);
```

Comment:
Why do you create list for `publicMachineKeys.size() + 1` but put there only one element `publicKeys.add(sshWorkspacePair.getPublicKey());`? Am I correct?